src/Symfony/Bridge/Twig/Extension/HttpFoundationExtension.php

```diff
@@ -39,8 +62,8 @@ public function __construct(RequestStack $requestStack, RequestContext $requestC
     public function getFunctions()
     {
         return [
-            new TwigFunction('absolute_url', [$this, 'generateAbsoluteUrl']),
-            new TwigFunction('relative_path', [$this, 'generateRelativePath']),
+            new TwigFunction('absolute_url', [$this->urlHelper, 'getAbsoluteUrl']),
+            new TwigFunction('relative_path', [$this->urlHelper, 'getRelativePath']),
```

Comment:
Please use a twig runtime, or a method on the extension, as that are the 2 ways which can compile to the most efficient code in Twig. Passing other callables like here requires initializing extensions to register all functions again, to have access to the callable. We don't do this in Symfony.

In this case, I suggest keeping the extensions method (and not deprecating them)